### ISSUE TITLE
Using :etag or :if-modified-since options do not work

### ISSUE THREAD

**Issue by olksdr** [clj-commons/tentacles#18]
After trying to work with this amazing library I wanted to relay on it fully in my small project. But what I found if there are few things which do not work as expected (using examples from the Readme):

* using `:etag` in any request does not work and just returns the same results over and over again 
```
=> (c/api-meta (repos/readme "Raynes" "tentacles" {}))
{:etag "W/\"c0aef320eab811abb09cf07fff8eb9a6\"", :last-modified "Wed, 26 Oct 2016 05:33:46 GMT", :call-limit 60, :call-remaining 56, :poll-interval nil}
=> (repos/readme "Raynes" "tentacles" {:etag "W/\"c0aef320eab811abb09cf07fff8eb9a6\""})
{:html_url "https://github.com/Raynes/tentacles/blob/master/README.md", :path "README.md", :encoding "base64", :git_url "https://api.github.com/repos/Raynes/tentacles/git/blobs/ad801aab92bb962d364aecf26dcabb
....... //// cut //// ....

=> (repos/readme "Raynes" "tentacles" {:etag "\"c0aef320eab811abb09cf07fff8eb9a6\""})
{:html_url "https://github.com/Raynes/tentacles/blob/master/README.md", :path "README.md", :encoding "base64", :git_url "https://api.github.com/repos/Raynes/tentacles/git/blobs/ad801aab92bb962d364aecf26dcabb2741fdf879", :content "[![Dependencies Status](https://jarkeeper.com/Raynes/tentacles/status.svg)](https://jarkeeper.com/Raynes/tentacles)\n\n# An octocat is nothing without her tentacles\n\nTentacles is a Clojure library for working with the Github v3 API. It supports the 
 ...... //// cut //// .....

=> 

```

* using `:if-modified-since` doesn't work either  with an error `Execution error (IllegalArgumentException) at tentacles.core/api-call$exec-request (core.clj:137).`
```
=> (c/api-meta (repos/readme "Raynes" "tentacles" {}))
{:etag "W/\"c0aef320eab811abb09cf07fff8eb9a6\"", :last-modified "Wed, 26 Oct 2016 05:33:46 GMT", :call-limit 60, :call-remaining 53, :poll-interval nil}
=> (repos/readme "Raynes" "tentacles" {:if-modified-since "Wed, 26 Oct 2016 05:33:46 GMT"})
Execution error (IllegalArgumentException) at tentacles.core/api-call$exec-request (core.clj:137).
Don't know how to create ISeq from: clojure.lang.Keyword


```

**Comment by MalloZup** [comment]
@olksdr thx for issue. AFAIK if you use the same `etag` over and over, it should expected behaviour that it return the same data. Imagine the e-tag as a key-of map, so if you acess with same key you will have same data.
The mechanism of e-tag, is that if the e-tag change, then you have new data.

**Comment by olksdr** [comment]
@MalloZup Thanks for the quick response. 

It's not entirely true: according to the github docs api https://developer.github.com/v3/#conditional-requests it returns nothing and no limits are consumed 

testing 
```
> curl https://api.github.com/users/olksdr -H 'If-None-Match: "2ee7841b34977b676941849cfff17f5e"'
> 

```

When I use the lib with examples above the results are returned and the rate limits are consumed.

**Comment by rborer** [comment]
@olksdr Thanks for your report. You're right that there seems to be an issue with the examples of code in the README. I'll try to look into it soon.

**Comment by rborer** [comment]
I've pushed a fix into #19. I haven't merged it yet as I still haven't figured out why ETag doesn't work (that is, if you pass what you get from github API it doesn't work)...

**Comment by olksdr** [comment]
> I've pushed a fix into #19. I haven't merged it yet as I still haven't figured out why ETag doesn't work (that is, if you pass what you get from github API it doesn't work)...

@rborer What I noticed when testing with `curl` that `Etag` in the `curl` output and after using `clj-http` are different even though I used the same URL (at the same time is `Last-Modified:` header are the same and correct). 

My understanding is that the `Etag` should be the same as well if one consumes the same resources.

With `curl` it looks like - `ETag: "2ee7841b34977b676941849cfff17f5e"`  (which is correct)
clj-http gives back this `"W/\"daad8acf7eaf63f2e1b9d35feeb0d6f5\""` (somehow encoded ?)

**Comment by rborer** [comment]
@olksdr Thanks, I noticed this difference too. the "W/" prefix indicate that it is a [weakly validating ETag](https://en.wikipedia.org/wiki/HTTP_ETag#Strong_and_weak_validation). Anyway, this is just a flag, the etag behaviour should be the same.

**Comment by olksdr** [comment]
@rborer 
Thanks! That explains a lot 😆 I was looking what does it mean and didn't manage to find (but it's actually so easy now)

so with next code it works now (using following headers `"Accept" "*/*"`)
```
> (client/get "https://api.github.com/users/olksdr" {:headers {"Accept" "*/*"}})

```
I'm getting correct  `Etag` -> ` "ETag" "W/\"2ee7841b34977b676941849cfff17f5e\""` and when I use it in curl it works as supposed to:

```
> curl https://api.github.com/users/olksdr -H 'If-None-Match: W/"2ee7841b34977b676941849cfff17f5e"' 
> 
```

**Comment by rborer** [comment]
Ok I found the issue, actually the headers were not passed properly. See commit 0b2d0be44fb5fe617489e099127fff60b76f8435 for more details.

I'll try to do a bit more tests before merging #19 and cutting a new release out.

**Comment by olksdr** [comment]
Nice, looks good to me. 
Thanks for digging into it!

**Comment by rborer** [comment]
Fix included in [release 0.6.4](https://github.com/clj-commons/tentacles/releases/tag/0.6.4).
